### PR TITLE
 Microsft.Web 2023-12-01: Add int32 format to all number types in the scaleAndConcurrency section of the functionAppConfig

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [x] Other, please clarify:
    
Update `2023-12-01` API version to add `"format": "int32"` to all number types in the scaleAndConcurrency section of the functionAppConfig.

In the latest `Microsoft.Web` API version, `2023-12-01`, I introduced the `functionAppConfig` property bag to specify configuration for a Flex Consumption function app. In the `scaleAndConcurrency` section, I defined four properties (`instanceCount`, `maximumInstanceCount`, `instanceMemoryMB`, and `perInstanceConcurrency`) where the expected value is an integer. However, I specified the type as `"type": "number"` but did not include the "format" property. As a result, the Java SDK defaults to creating a float, while our backend service expects an integer, causing the requests to fail. To fix this issue, I am adding `"format": "int32"` to each of the four properties. Please note that this is the first time we are generating SDKs that include the `functionAppConfig` property bag, which is why we would like to fix it in the latest API version `2023-12-01`.

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json [hunk 1]
```diff
@@ -3674,6 +3674,7 @@
         },
         "instanceCount": {
           "type": "number",
```
Comment (mikekistler) [inline specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json:3676]: Please make this change here and in the other three locations.
```suggestion
          "type": "integer",
```

**Review by mikekistler — CHANGES_REQUESTED** [review]
You should also change the `type` to `integer`.

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json (hunk 1 above)
Comment (Francisco-Gamino) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json]: Thank you, @mikekistler, for the review. Fixed.

**Comment by Francisco-Gamino** [comment]
Hello @mikekistler -- Are there any more changes required for this PR? If now, when can we merge it? Thank you.

**Comment by Francisco-Gamino** [comment]
> Hello @mikekistler -- Are there any more changes required for this PR? If not, when can we merge it? Thank you.

Thank you for the review @mikekistler. I will follow up with the SDK owners on the breaking changes.

**Comment by weidongxu-microsoft** [comment]
Java will take in the break, as previous `float` type won't be accepted by backend.

**Comment by Alancere** [comment]
Hi @raych1, please approve `BreakingChange-Go-Sdk-Suppression`, the current PR leads to the following `Go SDK BreakingChange`: (type form `*float32` to `*int32`)
```
- Type of `FunctionsAlwaysReadyConfig.InstanceCount` has been changed from `*float32` to `*int32`
- Type of `FunctionsScaleAndConcurrency.InstanceMemoryMB` has been changed from `*float32` to `*int32`
- Type of `FunctionsScaleAndConcurrency.MaximumInstanceCount` has been changed from `*float32` to `*int32`
- Type of `FunctionsScaleAndConcurrencyTriggersHTTP.PerInstanceConcurrency` has been changed from `*float32` to `*int32`
```

**Comment by Francisco-Gamino** [comment]
Hello @mikekistler -- The SDK breaking changes have been approved. Could you please approve this PR? Thank you.

**Review by mikekistler — APPROVED** [review]
Looks good. 👍

**Review by naveedaz — APPROVED** [review]

**Comment by Francisco-Gamino** [comment]
Hello @mikekistler -- Could you please merge this PR since we are not authorized to merge it? Thanks in advance.

**Comment by mikekistler** [comment]
@Francisco-Gamino I do not have merge permissions. You can post in the [Service API Toolset Teams channel](https://aka.ms/azsdk/support/spectools-channel) for help with merging.

**Comment by Francisco-Gamino** [comment]
> @Francisco-Gamino I do not have merge permissions. You can post in the [Service API Toolset Teams channel](https://aka.ms/azsdk/support/spectools-channel) for help with merging.

Will do. Thank you, @mikekistler.

**Pull request merged**